training_controller/prepare_training_logs.py

```diff
@@ -175,78 +147,126 @@ def fetch_training_logs(self, es_instance, num_logs_to_fetch, timestamps_list):
                 continue
+            start_ts, end_ts, filename = entry["start_ts"], entry["end_ts"], entry["filename"]
             current_command = esdump_sample_command[:]
             current_command[2] = current_command[2].format(
-                entry["start_ts"], entry["end_ts"]
+                start_ts, end_ts
             )
             current_command[6] = current_command[6].format(
                 timestamps_esdump_num_logs_fetched[idx]
             )
             current_command[10] = current_command[10].format(
-                os.path.join(self.es_dump_data_path, f"training_logs_{idx}.json")
+                os.path.join(self.ES_DUMP_DIR, f"{filename}")
             )
             query_queue.append(current_command)
         if len(query_queue) > 0:
             self.run_esdump(query_queue)
+            return True
+        else:
+            return False
 
-    def create_windows(self):
-        # For every json file, write/append each time window to own file
-        for es_split_json_file in os.listdir(self.es_dump_data_path):
-            if not ".json" in es_split_json_file:
-                continue
-            json_file_to_process = os.path.join(
-                self.es_dump_data_path, es_split_json_file
-            )
-            df = pd.read_json(json_file_to_process, lines=True)
-            df = pd.json_normalize(df["_source"])
-            df.groupby(["window_start_time_ns"]).apply(
-                lambda x: self.save_window(x.name, x)
-            )
-            # delete ESDumped file
-            os.remove(json_file_to_process)
-        shutil.rmtree(self.es_dump_data_path)
 
     def fetch_and_update_timestamps(self,es_instance):
         timestamps_list = []
         try:
             oldest_log = es_instance.search(index="logs", body={"aggs": {"min_ts": {"min": { "field": "timestamp"}}}, "_source": ["timestamp"]}, size=1)
+            newest_log = es_instance.search(index="logs", body={"aggs": {"max_ts": {"max": { "field": "timestamp"}}}, "_source": ["timestamp"]}, size=1)
         except Exception as e:
             logging.error(e)
             return timestamps_list
         oldest_log_timestamp = int(oldest_log["aggregations"]["min_ts"]["value"])
+        newest_log_timestamp = int(newest_log["aggregations"]["max_ts"]["value"])
         try:
             all_normal_intervals = scan(es_instance, index="opni-normal-intervals", query={"query": {"match_all": {}}})
         except Exception as e:
             logging.error("Error trying to retrieve all normal intervals from opni-normal-intervals index")
             return timestamps_list
         for normal_interval in all_normal_intervals:
             start_ts, end_ts = normal_interval["_source"]["start_ts"], normal_interval["_source"]["end_ts"]
+            full_file_prefix = "{}_{}".format(start_ts, end_ts)
+            interval_json_files = [
+                file
+                for file in os.listdir(self.TRAINING_DIR)
```

Comment:
you're running this static result as a for loop within another for loop
I suggest you os.listdir only once and store a set. then you can check to see if `full_file_prefix` is in this set